src/app/store/slices/plan/index.ts

```diff
@@ -1,152 +1,141 @@
-    if (user?.teams) {
-      promises.push(planService.fetchTeamPlan());
-    }
-
-    const [individualPlan, teamPlan] = await Promise.all(promises);
-
-    return { individualPlan, teamPlan };
-  },
-);
-
-export const fetchLimitThunk = createAsyncThunk<number, void, { state: RootState }>(
+export const fetchLimitThunk = createAsyncThunk<number, string | undefined, { state: RootState }>(
   'plan/fetchLimit',
-  async (payload: void, { getState }) => {
+  async (payload: string | undefined, { getState }) => {
     const isAuthenticated = getState().user.isAuthenticated;
     let limit = 0;
 
     if (isAuthenticated) {
-      limit = await limitService.fetchLimit();
+      limit = await limitService.fetchLimit(payload);
     }
 
     return limit;
   },
 );
 
-export const fetchUsageThunk = createAsyncThunk<UsageResponse | null, void, { state: RootState }>(
+export const fetchUsageThunk = createAsyncThunk<UsageResponse | null, string | undefined, { state: RootState }>(
   'plan/fetchUsage',
-  async (payload: void, { getState }) => {
+  async (payload: string | undefined, { getState }) => {
     const isAuthenticated = getState().user.isAuthenticated;
 
     if (isAuthenticated) {
-      return await usageService.fetchUsage();
+      return usageService.fetchUsage(payload);
     } else {
       return null;
     }
   },
 );
 
-export const fetchSubscriptionThunk = createAsyncThunk<UserSubscription | null, void, { state: RootState }>(
-  'plan/fetchSubscription',
-  async (payload: void, { getState }) => {
-    const isAuthenticated = getState().user.isAuthenticated;
-
-    if (isAuthenticated) {
-      return paymentService.getUserSubscription();
-    } else {
-      return null;
-    }
-  },
-);
+export const fetchSubscriptionThunk = createAsyncThunk<
+  UserSubscription | null,
+  { userType: UserType },
+  { state: RootState }
+>('plan/fetchSubscription', async (payload, { getState }) => {
+  const isAuthenticated = getState().user.isAuthenticated;
+
+  if (isAuthenticated) {
+    return paymentService.getUserSubscription(payload.userType);
+  } else {
+    return null;
+  }
+});
 
 export const planSlice = createSlice({
   name: 'plan',
   initialState,
   reducers: {
-    setSubscription: (state: PlanState, action: PayloadAction<UserSubscription>) => {
-      state.subscription = action.payload;
+    setSubscriptionIndividual: (state: PlanState, action: PayloadAction<UserSubscription>) => {
+      state.individualSubscription = action.payload;
+    },
+    setSubscriptionBusiness: (state: PlanState, action: PayloadAction<UserSubscription>) => {
+      state.businessSubscription = action.payload;
     },
   },
   extraReducers: (builder) => {
     builder
-      .addCase(initializeThunk.pending, () => undefined)
+      .addCase(initializeThunk.pending, (state) => {
+        state.businessPlan = null;
+        state.businessPlanLimit = 0;
+        state.businessPlanUsage = 0;
+        state.businessPlanUsageDetails = null;
```

Comment:
Have you made sure this doesn't break anything?
Because if you set everything to 0 every time you run this thunk and there is already data you will reset it and there might be some strange behaviour

Reply:
Yes. Every time we call the `initializeThunk` pertaining to plans, we are collecting the data again. In case of business, if we delete a business subscription, the data persists even if we call the `initializeThunk`. So, since we are collecting the data again and the business data is called by a condition (only if we have a workspace), it might be possible that the business data is not replaced and persisted.